controllers/discovery/initDiscoveryUserData.js

```diff
@@ -9,11 +9,30 @@ const {QueryTypes} = require('sequelize');
  * @returns
  */
 const InitDiscoveryUserData = async (req, res) => {
-  const {limit = 50, page = 0} = req.body;
+  const {limit = 50, page = 0} = req.query;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `limit` and `page` parameters are being extracted from `req.query`, which means they are expected to be passed as query parameters in the request URL. However, these values are not validated or sanitized before use. This could lead to potential SQL injection attacks if the values are directly used in SQL queries. It's also important to ensure that `limit` and `page` are positive integers.

```diff
-  const {limit = 50, page = 0} = req.query;
+  let {limit = 50, page = 0} = req.query;
+  limit = parseInt(limit);
+  page = parseInt(page);
+  if (isNaN(limit) || limit < 1) {
+    limit = 50; // default value
+  }
+  if (isNaN(page) || page < 0) {
+    page = 0; // default value
+  }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->